### DISCUSSION TITLE
How to add a custom query field?

### DISCUSSION THREAD

**Discussion by Diluka** [TriPSs/nestjs-query#131] [Q&A] [answered]
I want to have a parameter that can represent a complex set of query conditions. Query conditions that are difficult or impossible to implement with automatically generated parameters. 

Most of these custom queries are used as filters. Like `WHERE (generated) AND (custom)`

for example.
1. The query uses a join table that is too deep.
2. The query needs to be implemented using SQL functions
3. A virtual status query that requires all subitems to meet certain conditions.

**Comment by GP4cK** [comment]
You can use the same `@ResolveField()` as in nestjs. What's your use case?

**Reply by Diluka** [reply]
no, I know this is to add a schema mapping.

**Comment by TriPSs** [comment]
Could you elaborate a bit more?

**Reply by Diluka** [reply]
updated

**Comment by TriPSs** [comment, marked as answer]
I would recommend creating your own [resolver with a custom endpoint](https://tripss.github.io/nestjs-query/docs/graphql/resolvers#custom-endpoints) so you can extend the default query args, then when calling the service instead of doing the following:
```ts
    return TodoItemConnection.createFromPromise((q) => this.service.query(q), { ...query, ...{ filter } });
```
Do
```ts
    return TodoItemConnection.createFromPromise((q) =>  {
       const qbBuilder = this.service.filterQueryBuilder.select(q)
       // Do with qbBuilder what you want here
       return qbBuilder.getMany()
    }, { ...query, ...{ filter } });
```

**Reply by Diluka** [reply]
this looks good

**Reply by Diluka** [reply]
@TriPSs ~~`QueryService` has no property of `filterQueryBuilder`~~ I figure out it is `TypeOrmQueryService`

how to create `assemblers`?

If I want to add a new filter args on `todoItems` endpoint, what should I do?


And how about count query?

**Reply by Diluka** [reply]
Let me make the example a little more specific. I want to add a `searchText` parameter to the originally generated `todoItems` endpoint. Then implement a cross-field fuzzy search based on this parameter. Concatenate a query condition like `AND ( field1 LIKE :searchText OR field2 LIKE :searchText)` in both the query statement and the count statement. This will get a new array of query results and the total number of `totalCount` that satisfy the condition.

**Reply by Diluka** [reply]
It will cause null value error if I override `queryMany` in my custom `CRUDResolver`

```ts
// just like this will cause error
  async queryMany(query) {
    return super.queryMany(query);
  }
```


-----

I need copy all metadata from `super.constructor.prototype.queryMany` to `MyCustomResolver.prototype.queryMany`
